cmd/cloud-api-adaptor/main.go

```diff
@@ -185,6 +186,7 @@ func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 		})
 		defaultToEnv(&vspherecfg.UserName, "GOVC_USERNAME")
 		defaultToEnv(&vspherecfg.Password, "GOVC_PASSWORD")
+		defaultToEnv(&vspherecfg.Thumbprint, "GOVC_THUMBPRINT")
```

Comment:
entrypoint.sh also needs to be modified no?

Reply:
So I originally added it to entrypoint.sh and then saw in the CAA container logs that it was printed out 

+ exec cloud-api-adaptor-vsphere vsphere -vcenter-url [dhcp158-101.virt.lab.eng.bos.redhat.com](http://dhcp158-101.virt.lab.eng.bos.redhat.com/) -data-center Datacenter -data-store datastore2 -vxlan-port 9000 -template ocp-ubuntu-template -thumbprint 3D:2C:DE:3B:26:52:8C:58:C6:96:B1:1B:64:09:8B:22:EC:D9:DD:DD -socket /run/peerpod/hypervisor.sock

Which I did not want. If you notice UserName and Password are also handled in this way so they do not appear in the exec statement and GOVC_THUMBPRINT is optional.

exec cloud-api-adaptor-vsphere vsphere \
        -vcenter-url ${GOVC_URL}  \
        -data-center ${GOVC_DATACENTER} \
        -data-store ${GOVC_DATASTORE} \
        ${optionals} \
        -socket /run/peerpod/hypervisor.sock

Reply:
right, that's an environment var, didn't notice, thanks!